### DISCUSSION TITLE
Is it possible to use Python-SocketIO without bidict?

### DISCUSSION THREAD

**Discussion by navin-bhaskar** [miguelgrinberg/python-socketio#1342] [Q&A] [answered]
Bidict is released under MPL which requires distribution of source while SocketIO itself is under permissive MIT. Is it possible to have an alternate release that uses dic() instead of bidict() for rooms?

**Comment by miguelgrinberg** [comment]
I don't know what `dic()` is. If you know of any BSD/MIT alternatives I can have a look.

**Reply by navin-bhaskar** [reply]
Thanks for answering. Sorry for the mistake, I meant using in-built ditct(). Unfortunately I do not know of an alternative yet. 
Was bidict used purely for performance reasons? If that's the case, I think it would be nice to have a flavor of this lib that only uses permissively licensed variant albeit taking a performance hit.

**Reply by miguelgrinberg** [reply]
bidict is bidirectional. The regular dict is uni-directional.

**Reply by navin-bhaskar** [reply]
Right, I see 'ValueDuplicationError' being used to check if we already have room by the given name (if I am reading this correctly) we can do this check by have two dict()'s. Was this not considered for performance reasons or was there something else?

**Reply by miguelgrinberg** [reply, marked as answer]
I did not want to implement this myself, given that this package is available and performs the task really well. I'm not really sure what you are trying to get at, honestly. As I said above, if you have an alternative package that performs the same task with BSD/MIT license I can look into it. I'm not interested in implementing and maintaining a custom solution for this.
